crates/oxc_linter/src/rules/eslint/no_useless_call.rs

```diff
@@ -0,0 +1,207 @@
+use oxc_ast::{
+    ast::{CallExpression, ChainElement, Expression, MemberExpression},
+    match_member_expression, AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::cmp::ContentEq;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+fn no_useless_call_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Avoid unnecessary use of .call() and .apply()")
+        .with_help("Replace with a normal function invocation")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessCall;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Disallow unnecessary calls to .call() and .apply()
+    ///
+    /// ### Why is this bad?
+    /// Function.prototype.call() and Function.prototype.apply() are slower than the normal function invocation.
+    ///
```

Comment:
eslint tells about this part:

> This rule compares code statically to check whether or not thisArg is changed. So if the code about thisArg is a dynamic expression, this rule cannot judge correctly.

https://eslint.org/docs/latest/rules/no-useless-call#known-limitations

Reply:
Is it correct to add this under the "Why is this bad?" section?

Reply:
Yes. It would not always be seen when it is under "Examples"

Reply:
I’ve added to the "Why is this bad?" section in the commit below:
https://github.com/oxc-project/oxc/pull/8789/commits/5a58bea3a47b9065bd955c4c62e6ae9e160af195